core/trino-main/src/main/java/io/trino/execution/TableGetSplitDistribution.java

```diff
@@ -0,0 +1,40 @@
+package io.trino.execution;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.Distribution.DistributionSnapshot;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class TableGetSplitDistribution
+{
+    private final PlanNodeId planNodeId;
+    private final DistributionSnapshot splitDistribution;
+
+    @JsonCreator
+    public TableGetSplitDistribution(
+            @JsonProperty("planNodeId") PlanNodeId planNodeId,
+            @JsonProperty("SplitDistribution") DistributionSnapshot splitDistribution)
+    {
+        requireNonNull(planNodeId);
+        requireNonNull(splitDistribution);
+        this.planNodeId = planNodeId;
```

Comment:
`this.planNodeId = requireNotNull(planNodeId, "planNodeId is null")`.

Same for other field